### PR TITLE
Ports /vg/station drink sliding

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request
[EXAMPLE VIDEO](https://cdn.discordapp.com/attachments/336748540690825216/673640806673809429/idw3gh.mp4)

Ports [/vg/station's drink farsliding](https://github.com/vgstation-coders/vgstation13/pull/24915) by ShiftyRail ~as a replacement for TG's drink flinging. Keeps the book and action for flinging but it's now used for sliding.~  as an additional ability granted from drink flinging. The bartender starts with the ability now instead of having to read the book.

What this does is allow people with the flinging ability to ~instead~ slide drinks across straight lines of tables (up to 7 tiles away) by standing next to the drink and dragging it to the target table. People without the flinging ability can slide drinks by one tile with a chance of failure. The example video at the top of this description demonstrates it in action.

~Meant as an alternative to #10766~ Changed to not remove any existing functionality.

## Why It's Good For The Game

it's a cool~er~ feature imo

## Changelog
:cl:
add: Ported drink sliding from /vg/station.
/:cl:

### CODE REVIEW

**Comment by Linzolle** [comment]
example video is not loading?

**Comment by ghost** [comment]
should work now

**Comment by Owai-Seek** [comment]
From what I'm hearing people like drink flinging because they can pound out several drinks at the bar machine and just toss them, this is nice for RP, but mechanic-wise its not what people were upset about.

**Comment by silicons** [comment]
neat feature but i don't think it should replace drink throws.

**Comment by ghost** [comment]
I'm not entirely happy with my implementation here so I'm probably going to rework this so as not to replace the original mechanic

File: code/__HELPERS/vector.dm [hunk 1]
```diff
@@ -0,0 +1,57 @@
+// Basic geometry things.
+
+/vector/
```
Comment (Ghommie) [inline code/__HELPERS/vector.dm]: I assume this is a datum, so could you state the full path here and below if that's true, please?
```suggestion
/datum/vector
```

File: code/modules/food_and_drinks/drinks/drinks.dm [hunk 2]
```diff
@@ -126,12 +126,74 @@
...
+	var/mob/user = usr
+	if (!istype(src_location))
+		return
+	if (!user || user.incapacitated())
+		return
+	// Attempted drink sliding
+	if (locate(/obj/structure/table) in src_location)
+		//Are we an expert slider?
+		var/datum/action/innate/A = get_action_of_type(user, /datum/action/innate/drink_fling)
+		if(A?.active)
```
Comment (Ghommie) [inline code/modules/food_and_drinks/drinks/drinks.dm]: Could you use some early return logic for indentation, if possible?

File: code/modules/food_and_drinks/drinks/drinks.dm [hunk 3]
```diff
@@ -126,12 +126,74 @@
...
 	qdel(src)
 
+/obj/item/reagent_containers/food/drinks/MouseDrop(atom/over, atom/src_location, atom/over_location, src_control, over_control, params)
+	var/mob/user = usr
+	if (!istype(src_location))
+		return
+	if (!user || user.incapacitated())
+		return
+	// Attempted drink sliding
+	if (locate(/obj/structure/table) in src_location)
```
Comment (Ghommie) [inline code/modules/food_and_drinks/drinks/drinks.dm]: Same here.

**Review by Ghommie — CHANGES_REQUESTED** [review]
Just a few little things, overall good.

File: code/__HELPERS/vector.dm (hunk 1 above)
Comment (ghost) [reply code/__HELPERS/vector.dm]: done

File: code/modules/food_and_drinks/drinks/drinks.dm (hunk 2 above)
Comment (ghost) [reply code/modules/food_and_drinks/drinks/drinks.dm]: done as best I can without breaking up the proc

**Review by silicons — APPROVED** [review]
:+1: from me since drink throwing is back in, once ghom's stuff is done

**Review by Ghommie — APPROVED** [review]

**Pull request merged**